### PR TITLE
feature/loaded-csv-files

### PR DESCRIPTION
**Pull request by Amisha121102**
Written a script to load csv files and connected that to database

### CODE REVIEW

File: phase1/amisha/main.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+def main():
+    user = 'root'
+    password = 'root'
+    host = 'localhost'
+    database = 'FincityData3'
+    engine = create_engine(f'mysql+mysqlconnector://{user}:{password}@{host}/{database}')
+    
+    files = get_table_name()
+    for file in files:
+        table_name=file.split("\\")[-1].split(".")[0]
```
Comment (lawbringr1) [inline phase1/amisha/main.py:14]: what will happen we are using UNIX path? using /

File: phase1/amisha/utils.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        try:
+            chunk.to_sql(name=table_name, con=engine, if_exists='append', index=False)
+            logging.info(f'Inserted chunk {i} with {len(chunk)} rows')
+            i += 1
+        except SQLAlchemyError as e:
+            logging.error(f"Error inserting data: {e}")
+            break
+
+def get_table_name():
+    main_folder = 'phase1\\amisha\\Databases'
```
Comment (lawbringr1) [inline phase1/amisha/utils.py:33]: what will happen we are using UNIX path? using /

File: phase1/amisha/utils.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    folders = [name for name in os.listdir(main_folder) if os.path.isdir(os.path.join(main_folder, name))]
+ 
+    database=folders[0]
+    database_path=main_folder+"\\"+database
```
Comment (lawbringr1) [inline phase1/amisha/utils.py:37]: what will happen we are using UNIX path? using /

File: phase1/amisha/utils.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+            i += 1
+        except SQLAlchemyError as e:
+            logging.error(f"Error inserting data: {e}")
+            break
+
+def get_table_name():
+    main_folder = 'phase1\\amisha\\Databases'
+    folders = [name for name in os.listdir(main_folder) if os.path.isdir(os.path.join(main_folder, name))]
+ 
+    database=folders[0]
```
Comment (lawbringr1) [inline phase1/amisha/utils.py:36]: if folder is not there throw error

File: phase1/amisha/utils.py (hunk 3 above)
Comment (lawbringr1) [inline phase1/amisha/utils.py:37]: why are you taking database_path that is already there right?

File: phase1/amisha/main.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
+from sqlalchemy import create_engine
+from utils import create_table, get_table_name, insert_data
+
+
+def main():
+    user = 'root'
```
Comment (lawbringr1) [inline phase1/amisha/main.py:6]: dont send any sensitive info to github

File: phase1/amisha/schema.py [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import pandas as pd
+from sqlalchemy import TEXT, BigInteger, Column, DateTime, Float, Integer, String
+
+
+def schemaCreator(sample_chunk):
```
Comment (lawbringr1) [inline phase1/amisha/schema.py:5]: use python naming conventions

File: phase1/amisha/utils.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+        try:
+            chunk.to_sql(name=table_name, con=engine, if_exists='append', index=False)
+            logging.info(f'Inserted chunk {i} with {len(chunk)} rows')
+            i += 1
+        except SQLAlchemyError as e:
+            logging.error(f"Error inserting data: {e}")
+            break
+
+def get_table_name():
+    main_folder = 'phase1/amisha/Databases'
```
Comment (lawbringr1) [inline phase1/amisha/utils.py:34]: what will happen if path uses \ ?

File: phase1/amisha/utils.py [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+    return table
+
+def insert_data(csv_file_path, table_name, engine, chunk_size=15000):
+    i = 0
+    for chunk in pd.read_csv(csv_file_path, chunksize=chunk_size):
+        try:
+            chunk.to_sql(name=table_name, con=engine, if_exists='append', index=False)
+            logging.info(f'Inserted chunk {i} with {len(chunk)} rows')
+            i += 1
+        except SQLAlchemyError as e:
```
Comment (lawbringr1) [inline phase1/amisha/utils.py:29]: what is e?

Comment (Amisha121102) [reply phase1/amisha/utils.py:29]: Can I changed it to "exception"?

File: phase1/amisha/utils.py [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+    try:
+        metadata.create_all(engine)
+        logging.info(f"Table '{table_name}' is created or already exists.")
+    except SQLAlchemyError as e:
+        logging.error(f"Error creating table: {e}")
+    return table
+
+def insert_data(csv_file_path, table_name, engine, chunk_size=15000):
+    i = 0
+    for chunk in pd.read_csv(csv_file_path, chunksize=chunk_size):
```
Comment (lawbringr1) [inline phase1/amisha/utils.py:24]: use enumurate here instead of i

File: phase1/amisha/main.py [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+from sqlalchemy import create_engine
+from utils import create_table, get_table_name, insert_data
+
+
+def main():
+    user = '****'
+    password = '****'
+    host = '****'
+    database = 'FincityData'
+    engine = create_engine(f'mysql+mysqlconnector://{user}:{password}@{host}/{database}')
```
Comment (lawbringr1) [inline phase1/amisha/main.py:11]: you can also move engine creation to a function

File: phase1/amisha/utils.py (hunk 8 above)
Comment (lawbringr1) [reply phase1/amisha/utils.py:29]: sql_alchemy_exception

File: phase1/amisha/utils.py [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+        try:
+            chunk.to_sql(name=table_name, con=engine, if_exists='append', index=False)
+            logging.info(f'Inserted chunk {i} with {len(chunk)} rows')
+            i += 1
+        except SQLAlchemyError as e:
+            logging.error(f"Error inserting data: {e}")
+            break
+
+def get_table_name():
+    main_folder = os.path.join("phase1", "amisha")
```
Comment (lawbringr1) [inline phase1/amisha/utils.py:34]: you can get current directory so you don't need to create path like this

File: phase1/amisha/utils.py (hunk 8 above)
Comment (Amisha121102) [reply phase1/amisha/utils.py:29]: okay!

File: phase1/amisha/utils.py [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+    main_folder_path = Path(os.path.join(main_folder,'Databases'))
+    if not main_folder_path.exists():
+        logging.error(f"The directory '{main_folder}' does not exist.")
```
Comment (lawbringr1) [inline phase1/amisha/utils.py:37]: throw error here instead of continuing

**Review by lawbringr1 — APPROVED** [review]
Nice Pr

**Pull request merged**